python_modules/libraries/dagster-aws/dagster_aws_tests/s3_tests/test_s3_file_manager.py

```diff
@@ -207,3 +209,62 @@ def test_pipeline():
 
     execute_pipeline(test_pipeline)
     assert did_it_run["it_ran"]
+
+@mock.patch("boto3.session.Session.resource")
+@mock.patch("dagster_aws.s3.resources.S3FileManager")
+def test_s3_file_manager_resource_with_profile(MockS3FileManager, mock_boto3_resource):
+    did_it_run = dict(it_ran=False)
+
+    resource_config = {
+        "use_unsigned_session": True,
+        "region_name": "us-west-1",
+        "endpoint_url": "http://alternate-s3-host.io",
+        "s3_bucket": "some-bucket",
+        "s3_prefix": "some-prefix",
+        "profile_name": "some-profile"
+    }
+
+    mock_s3_session = mock_boto3_resource.return_value.meta.client
+
+    @solid(required_resource_keys={"file_manager"})
+    def test_solid(context):
+        # test that we got back a S3FileManager
```

Comment:
The body of `test_solid` can be empty. As you pointed out, we expect this to fail during resource initialization before anything in `test_solid` actually runs. In the interest of simplicity, we should remove any lines of code that aren't directly related to the behavior we're testing.

It's a bit of an unfortunate situation that we need to set up boilerplate solid and pipeline definitions only to test resource initialization - our core team is working on an improvement for 0.12.0 that will allow us to initialize resources directly for tests.

Reply:
Oh okay, cool. I will add assertion as you have suggested, thanks!